pkg/collector/corechecks/system/winregistry/winregistry.go

```diff
@@ -0,0 +1,191 @@
+//go:build windows
+
+package winregistry
+
+import (
+	"errors"
+	"fmt"
+	"io/fs"
+	"strconv"
+	"strings"
+
+	"github.com/DataDog/datadog-agent/pkg/autodiscovery/integration"
+	"github.com/DataDog/datadog-agent/pkg/collector/check"
+	core "github.com/DataDog/datadog-agent/pkg/collector/corechecks"
+	"github.com/DataDog/datadog-agent/pkg/metrics"
+	"github.com/DataDog/datadog-agent/pkg/util"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+	"golang.org/x/sys/windows/registry"
+	"gopkg.in/yaml.v2"
+)
+
+const (
+	checkName   = "windows_registry" // this appears in the Agent Manager and Agent status
+	checkPrefix = "winregistry"      // This is the prefix used for all metrics emitted by this check
+)
+
+type metricCfg struct {
+	Name           string
+	Mappings       []map[string]float64   `yaml:"mapping"`
+	ValueIfMissing util.Optional[float64] `yaml:"value_if_missing, omitempty"`
+}
+
+type registryKeyCfg struct {
+	Name    string
+	Metrics map[string]metricCfg
+}
+
+type checkCfg struct {
+	RegistryKeys map[string]registryKeyCfg `yaml:"registry_keys"`
+}
+
+// registryKey is the in-memory representation of the key to monitor
+// it's different from the regstryKeyCfg because we need to split the hive
+// from the keypath. It's easier to do this once during the Configure phase,
+// than every time the check runs.
+type registryKey struct {
+	name            string
+	hive            registry.Key
+	keyPath         string
+	originalKeyPath string // keep the original keypath around, for logging errors
+	metrics         map[string]metricCfg
+}
+
+// WindowsRegistryCheck contains the field for the WindowsRegistryCheck
+type WindowsRegistryCheck struct {
+	core.CheckBase
+	metrics.Gauge
+	registryKeys []registryKey
+}
+
+func (c *WindowsRegistryCheck) Configure(integrationConfigDigest uint64, data integration.Data, initConfig integration.Data, source string) error {
+	err := c.CommonConfigure(integrationConfigDigest, initConfig, data, source)
+	if err != nil {
+		return err
+	}
+
+	var conf checkCfg
+	if err := yaml.Unmarshal(data, &conf); err != nil {
+		return err
+	}
+
+	hiveMap := map[string]registry.Key{
+		"HKLM":               registry.LOCAL_MACHINE,
+		"HKEY_LOCAL_MACHINE": registry.LOCAL_MACHINE,
+		"HKU":                registry.USERS,
+		"HKEY_USERS":         registry.USERS,
+		"HKEY_CLASSES_ROOT":  registry.CLASSES_ROOT,
+		"HKCR":               registry.CLASSES_ROOT,
+	}
+
+	for regKey, regKeyConfig := range conf.RegistryKeys {
+		splitKeypath := strings.Split(regKey, "\\")
+		if len(splitKeypath) < 2 {
+			return log.Errorf("the key %s is too short to be a valid key", regKey)
```

Comment:
If you want to log the error you might want to add some context (eg. the name of the check)
[Same comment below]

Reply:
> I'm curious, I did not see any other checks that put their check name in their logs - would you have an example to share ?

Honestly no it was just a feeling haha
I just thought that if I saw "the key ... is too short" I'd have no idea what it's about, but I guess since there are the source file and line of the log it's not really the case

> Also, I wonder what's better in this case:
> - Return log.Errorf or
> - Print the error to the log and continue loading the check.

I'll ask my team what they think about it

Reply:
Everyone who answered said they preferred returning an error early, but it was pointed out that the agent often does the other way around.

As consistency is important in the agent, you might want to ask agent-integrations what they usually do in this kind of situation.